### PR TITLE
Move integration tests from unit-testing into integration-testing

### PR DESCRIPTION
**Pull request by ericvergnaud**
Python unit tests currently comprise integration tests that are slow or impossible to run OOTB from a desktop.
This PR solves the issue by:
- moving integration tests from the `units` hierarchy to the `integration` one
- adding a `python-integration` job to the CI that runs tests under `integration`

### CODE REVIEW

**Comment by ericvergnaud** [comment]
@gueniai You might want to add `integration-python` to the list of `required` successful ci jobs

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -48,6 +48,37 @@ jobs:
...
+        uses: actions/setup-python@v5
+        with:
+          cache: 'pip'
+          cache-dependency-path: '**/pyproject.toml'
+          python-version: '3.10'
+
+      - name: Install hatch
+        run: pip install hatch==$HATCH_VERSION
+
+      - name: Setup Spark Remote
```
Comment (sundarshankar89) [inline .github/workflows/push.yml:67]: Can you create a step in make file as well?

**Review by sundarshankar89 — COMMENTED** [review]
Small NIt: rest LGTM

**Review by bishwajit-db — APPROVED** [review]
LGTM

**Comment by sundarshankar89** [comment]
@gueniai  need you to make the integration test mandatory.

File: .github/workflows/push.yml (hunk 1 above)
Comment (ericvergnaud) [reply .github/workflows/push.yml:67]: it already exists

**Comment by sundarshankar89** [comment]
@ericvergnaud created https://github.com/databrickslabs/remorph/pull/1377 to add what I was talking about.

**Review by sundarshankar89 — APPROVED** [review]
LGTM
Tackled in https://github.com/databrickslabs/remorph/pull/1377

**Pull request merged**